napari/layers/_base_layer/model.py

```diff
@@ -92,6 +95,39 @@ def name(self, name):
         self._name = name
         self.events.name()
 
+    @property
+    def indices(self):
+        """Tuple of slice objects for slicing arrays on each dimension."""
+        return self._indices
+
+    @indices.setter
+    def indices(self, indices):
+        if indices == self.indices:
+            return
+        self._indices = indices[-self.ndim:]
+        self._set_view_slice()
+
+    @property
+    def cursor_position(self):
+        """tuple: 2-tuple of cursor in image coordinates. Setter does
+        transformation from canvas coordinates to image coordinates
+        """
+        return self._cursor_position
+
+    @cursor_position.setter
+    def cursor_position(self, cursor_position):
+        transform = self._node.canvas.scene.node_transform(self._node)
+        self._cursor_position = tuple(transform.map(cursor_position)[:2])
+
+    @property
+    def coordinates(self):
+        """tuple: Tuple of floats for slicing arrays on each dimension at
+        the current cursor position."""
```

Comment:
In this docstring, or maybe as comments, it is again worthwhile to explain the different coordinate conventions between vispy and NumPy, and that that explains the swapping of the dimension ordering in the following lines.

Reply:
And, it might be a good idea, whenever coordinates are mentioned *anywhere* to specify whether they are in canvas units and/or ordering, or in pixel/array units and/or ordering.

Reply:
... And perhaps we should aim to have a *single* place where the two are converted, as currently it seems to happen all over the place, which might make refactors and modifications (such as arbitrary ordering/display) harder than necessary. Does this seem feasible to you in the short term?

Reply:
yup, agreed - I'm cleaning this up, and will improve the doc strings